File: runner/languages/haskell/runner.hs
import System.IO
import System.Environment
import Task

main = do
    args <- getArgs
    let task = head args
    let test = head (tail args)

    let fileToRead = if test == "test" then "test.in" else "solve.in"

    fileContents <- readFile fileToRead
    let fileLines = lines fileContents

    let result = if task == "1" then taskOne fileLines else taskTwo fileLines

    putStrLn result

File: runner/languages/typescript/runner.ts
import { readFile } from "fs/promises";
import { taskOne, taskTwo } from "./task";

async function main() {
    let lastArgument = process.argv.pop() as string;
    let taskNumber = 1;
    let isTest = false;

    if (lastArgument === "test") {
        isTest = true;
        taskNumber = parseInt(process.argv.pop() as string);
    } else {
        taskNumber = parseInt(lastArgument)
    }

    const fileToLoad = isTest ? "test.in" : "solve.in";
    const fileContents = await readFile(fileToLoad, "utf-8")
    
    const lines = fileContents.trimEnd().split("\n");

    if (taskNumber === 1) {
        await taskOne(lines);
    }
    if (taskNumber === 2) {
        await taskTwo(lines);
    }
}

void main();